backend/app/workers/inference.py

```diff
@@ -71,10 +76,14 @@ def on_inference_completed(inf_result: DetectionResult, userdata: dict[str, Any]
                     queue_data = frame_queue.get(timeout=1)
                 except queue.Empty:
                     continue
+
+                inference_start_time = metrics_collector.record_inference_start()
                 model.infer_async(
                     queue_data.frame_data,
                     user_data={"stream_data": queue_data, "model_name": model_service.get_active_model_name()},
                 )
+                model_service.get_loaded_inference_model()
+                metrics_collector.record_inference_end(model_id=loaded_model.id, start_time=inference_start_time)
```

Comment:
This is not the right place to call `record_inference_end`: the inference is asynchronous, so at this point the inference request has been submitted but you have no guarantee that the inference itself is complete. Instead, you should compute the duration in the callback (although not very accurate due to scheduling overheads); you can pass the start time to the callback via `infer_async`.